### PR TITLE
Added SML layer

### PR DESCRIPTION
**Pull request by Devagamster**
Added a simple SML layer which interacts with Standard ML of New Jersey. Adds some basic REPL interaction bindings and a couple electric key functions.

### CODE REVIEW

File: contrib/lang/sml/packages.el [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+;; Copyright (c) 2014-2015 Keith Simmons & Contributors
+;;
+;; Author: Keith Simmons <keith@the-simmons.net>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(defvar sml-packages
```
Comment (trishume) [inline contrib/lang/sml/packages.el]: This should be `setq` so reloading works properly. It's a new thing.

File: contrib/lang/sml/packages.el [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    )
+  "List of all packages to install and/or initialize. Built-in packages
+which require an initialization must be listed explicitly in the list.")
+
+(defvar sml-excluded-packages '()
+  "List of packages to exclude.")
+
+(defun sml/init-sml-mode ()
+  (use-package sml-mode
+    :mode ("\\.\\(sml\\|sig\\)\\'" . sml-mode)
```
Comment (jonsterling) [inline contrib/lang/sml/packages.el]: Perhaps also `.fun`?

File: contrib/lang/sml/packages.el [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+;; Copyright (c) 2014-2015 Keith Simmons & Contributors
+;;
+;; Author: Keith Simmons <keith@the-simmons.net>
+;; URL: https://github.com/syl20bnr/spacemacs
+;;
+;; This file is not part of GNU Emacs.
+;;
+;;; License: GPLv3
+
+(setq sml-packages
```
Comment (jonsterling) [inline contrib/lang/sml/packages.el]: When I tested this locally using the latest version of Spacemacs, I got a "type error" referring to the docstring at the end of this definition. Removing the `"List of all...."` argument fixed it.

**Comment by syl20bnr** [comment]
Thank you ! :+1: 
I may various changes and fixes in commit: https://github.com/syl20bnr/spacemacs/commit/2251674

Cherry-picked in `develop` branch, you can safely delete your branch.

**Pull request closed without merging**